pkg/operator/controller/ingress/controller_test.go

```diff
@@ -994,26 +994,25 @@ func TestTLSProfileSpecForSecurityProfile(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		ic := &operatorv1.IngressController{
-			Spec: operatorv1.IngressControllerSpec{
-				TLSSecurityProfile: tc.profile,
-			},
-		}
-		tlsProfileSpec := tlsProfileSpecForSecurityProfile(ic.Spec.TLSSecurityProfile)
-		err := validateTLSSecurityProfile(ic)
-		if tc.valid && err != nil {
-			t.Errorf("%q: unexpected error: %v\nprofile:\n%s", tc.description, err, util.ToYaml(tlsProfileSpec))
-			continue
-		}
-		if !tc.valid && err == nil {
-			t.Errorf("%q: expected error for profile:\n%s", tc.description, util.ToYaml(tlsProfileSpec))
-			continue
-		}
-		if tc.expectedSpec != nil && !reflect.DeepEqual(tc.expectedSpec, tlsProfileSpec) {
-			t.Errorf("%q: expected profile:\n%s\ngot profile:\n%s", tc.description, util.ToYaml(tc.expectedSpec), util.ToYaml(tlsProfileSpec))
-			continue
-		}
-		t.Logf("%q: got expected values; profile:\n%s\nerror value: %v", tc.description, util.ToYaml(tlsProfileSpec), err)
+		t.Run(tc.description, func(t *testing.T) {
+			ic := &operatorv1.IngressController{
+				Spec: operatorv1.IngressControllerSpec{
+					TLSSecurityProfile: tc.profile,
+				},
+			}
+			tlsProfileSpec := tlsProfileSpecForSecurityProfile(ic.Spec.TLSSecurityProfile)
+			err := validateTLSSecurityProfile(ic)
+			if tc.valid && err != nil {
+				t.Fatalf("unexpected error: %v\nprofile:\n%s", err, util.ToYaml(tlsProfileSpec))
```

Comment:
Added this update after reviewing ,original commit was missing removal of tc.description.